### DISCUSSION TITLE
Is const reference always thread safe when used as function parameter? | Bhavith C

### DISCUSSION THREAD

**Discussion by giscus** [bhavithc/bhavithc.com#22] [Q&A]
# Is const reference always thread safe when used as function parameter? | Bhavith C

Many people has a fear of thread safety when using const reference as parameter to a function. As usual lets start with some examples and understand the thread safety of const reference

https://bhavithc.com/posts/is-const-reference-always-thread-safe-when-used-as-function-parameter/

**Comment by MahajanSamiksha** [comment]
The behavior of shared_ptr will be same as that of reference because with shared_ptr too, a new object is created every-time object is passed in the function
